File: backend/versuion/src/main/java/com/example/versuion/models/MvtStk.java
package com.example.versuion.models;

import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.Instant;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
public class MvtStk extends AbstractEntity{

    private Instant dateMvt;

    private BigDecimal quantite;

     @ManyToOne
    @JoinColumn(name = "idearticle")
    private Article article;

    @Enumerated(EnumType.STRING)
    private TypeMvtStk typeMvt;

    @Column(name = "sourcemvt")
    @Enumerated(EnumType.STRING)
    private SourceMvtStk sourceMvt;

    @Column(name = "identreprise")
    private Integer idEntreprise;

}

File: frontend/my-first-project/src/app/Models/MvtStk.ts
import { Article } from "./Article";

export interface MvtStk {

  id?: number;
  dateMvt?: number;
  quantite?: number;
  article?: Article;
  typeMvt?: 'ENTREE' | 'SORTIE' | 'CORRECTION_POS' | 'CORRECTION_NEG';
  sourceMvt?: 'COMMANDE_CLIENT' | 'COMMANDE_FOURNISSEUR' | 'VENTE';
  idEntreprise?: number;
}
